### PR TITLE
[GPU] concatenation_onednn migration

### PR DESCRIPTION
**Pull request by hyunback**
- update concatenation_onednn
- add unit test using onednn concat

Signed-off-by: hyunback <hyunback.kim@intel.com>

### CODE REVIEW

File: inference-engine/thirdparty/clDNN/tests/test_cases/concatenation_gpu_test.cpp [hunk 1]
```diff
@@ -783,3 +783,77 @@ INSTANTIATE_TEST_SUITE_P(smoke_low_precision,
...
+            concatenation("concat",
+                          { "input0", "input1", "input2", "input3", "input4" },
+                          concatenation::concatenation_axis::along_f,
+                          data_types::f32,
+                          "",
+                          padding{ { 0,0,0,0 }, 0 })
+    );
+
+    build_options options_target;
+    options_target.set_option(build_option::outputs({ "concat"}));
```
Comment (lznamens) [inline inference-engine/thirdparty/clDNN/tests/test_cases/concatenation_gpu_test.cpp:826]: Please use a unified code style and add a space: `({ "concat" })`

File: inference-engine/thirdparty/clDNN/tests/test_cases/concatenation_gpu_test.cpp [hunk 2]
```diff
@@ -783,3 +783,77 @@ INSTANTIATE_TEST_SUITE_P(smoke_low_precision,
...
+    implementation_desc impl = {format::bfyx, std::string(""), impl_types::onednn};
```
Comment (lznamens) [inline inference-engine/thirdparty/clDNN/tests/test_cases/concatenation_gpu_test.cpp:827]: The same.

File: inference-engine/thirdparty/clDNN/tests/test_cases/concatenation_gpu_test.cpp [hunk 3]
```diff
@@ -783,3 +783,77 @@ INSTANTIATE_TEST_SUITE_P(smoke_low_precision,
...
+    options_target.set_option(build_option::force_implementations({{"concat", impl}}));
```
Comment (lznamens) [inline inference-engine/thirdparty/clDNN/tests/test_cases/concatenation_gpu_test.cpp:828]: The same.

**Review by lznamens — DISMISSED** [review]

File: inference-engine/thirdparty/clDNN/src/impls/onednn/concatenation_onednn.cpp [hunk 4]
```diff
@@ -18,10 +18,110 @@ namespace cldnn {
 namespace onednn {
```
Comment (isanghao) [inline inference-engine/thirdparty/clDNN/src/impls/onednn/concatenation_onednn.cpp:20]: nit: please remove one space line here

File: inference-engine/thirdparty/clDNN/src/layout_optimizer.cpp [hunk 5]
```diff
@@ -884,6 +884,16 @@ impl_types layout_optimizer::get_preferred_impl_type(program_node& node) {
         if (input_layout.format.dimension() != output_layout.format.dimension()) {
             preferred_impl = impl_types::ocl;
         }
+    } else if (node.is_type<concatenation>()) {
+        if (!node.get_program().get_engine().get_device_info().supports_immad)
```
[the remark this answers is not in the document]
Comment (isanghao) [reply inference-engine/thirdparty/clDNN/src/layout_optimizer.cpp:888]: Now use_onednn_impl is implemented. Please use it: https://github.com/openvinotoolkit/openvino/pull/7716

File: inference-engine/thirdparty/clDNN/src/impls/onednn/concatenation_onednn.cpp [hunk 6]
```diff
@@ -18,10 +18,110 @@ namespace cldnn {
...
+        return make_unique<concatenation_onednn>(*this);
+    }
+
+    std::unordered_map<int, dnnl::memory> get_arguments(concatenation_inst& instance) const override {
+        std::unordered_map<int, dnnl::memory> args;
+
+        int input_idx = DNNL_ARG_MULTIPLE_SRC;
+        for (size_t i = 0; i < instance.inputs_memory_count(); i++) {
+            auto& input = instance.input_memory(i);
+            args.insert({input_idx++, input.get_onednn_memory(_pd.src_desc(static_cast<int>(i))) });
```
[the remark this answers is not in the document]
Comment (hyunback) [reply inference-engine/thirdparty/clDNN/src/impls/onednn/concatenation_onednn.cpp]: Fixed

File: inference-engine/thirdparty/clDNN/src/layout_optimizer.cpp (hunk 5 above)
[the remark this answers is not in the document]
Comment (hyunback) [reply inference-engine/thirdparty/clDNN/src/layout_optimizer.cpp]: Use use_onednn_impl

File: inference-engine/thirdparty/clDNN/tests/test_cases/concatenation_gpu_test.cpp (hunk 2 above)
Comment (hyunback) [reply inference-engine/thirdparty/clDNN/tests/test_cases/concatenation_gpu_test.cpp]: Fixed

File: inference-engine/thirdparty/clDNN/tests/test_cases/concatenation_gpu_test.cpp (hunk 1 above)
Comment (hyunback) [reply inference-engine/thirdparty/clDNN/tests/test_cases/concatenation_gpu_test.cpp]: Fixed

File: inference-engine/thirdparty/clDNN/tests/test_cases/concatenation_gpu_test.cpp (hunk 3 above)
Comment (hyunback) [reply inference-engine/thirdparty/clDNN/tests/test_cases/concatenation_gpu_test.cpp]: Fixed

File: inference-engine/thirdparty/clDNN/src/impls/onednn/concatenation_onednn.cpp (hunk 4 above)
Comment (hyunback) [reply inference-engine/thirdparty/clDNN/src/impls/onednn/concatenation_onednn.cpp]: Fixed

**Review by isanghao — APPROVED** [review]

**Pull request merged**